cmd/crdvalidator/handlers/crd.go

```diff
@@ -73,3 +78,20 @@ func (cv *CrdValidator) InjectDecoder(d *admission.Decoder) error {
 	cv.decoder = d
 	return nil
 }
+
+// enabled takes a CRD and checks its content to see if it is is owned
+// by RukPak or if crdvalidator is disabled explicitly
+func enabled(crd *apiextensionsv1.CustomResourceDefinition) bool {
+	// Check if the annotation is set explicitly
+	if annotation, ok := crd.GetAnnotations()[crdanno.ValidationKey]; ok {
+		return annotation == crdanno.Enabled
+	}
+
+	// Read the owner reference and check to see if it is owned by RukPak
+	for _, ref := range crd.OwnerReferences {
+		if ref.APIVersion == rukpakv1alpha1.GroupVersion.String() {
+			return true
+		}
+	}
```

Comment:
I think I'd prefer if we check for the presence of known labels used by the controllers instead of owner references.

Testing this locally with a Bundle/BI that references the main branch of the combo repository, we should be in the clear to use label selectors:

```console
$ kubectl get crds --show-labels
combinations.combo.io                 2022-04-28T22:06:22Z   app.kubernetes.io/managed-by=Helm,core.rukpak.io/owner-kind=BundleInstance,core.rukpak.io/owner-name=combo
```

Where the relevant labels are `core.rukpak.io/owner-kind=BundleInstance,core.rukpak.io/owner-name=combo`.

Reply:
I went ahead and updated the manifests for the webhook to only receive CRDs that are owned by RukPak. If that CRD has the `core.rukpak.io/safe-upgrade-validation` annotation set to `disabled` it will not validate it. Otherwise, the validation logic will run.